service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java

```diff
@@ -128,4 +141,20 @@ private void setServiceProvider(Encounter encounter, POCDMT000002UK01ClinicalDoc
         encounter.setServiceProvider(serviceProvider);
         encounter.setServiceProviderTarget(serviceProviderOrganization);
     }
+
+    private void setSubject(Encounter encounter, POCDMT000002UK01ClinicalDocument1 clinicalDocument1) {
+        if (clinicalDocument1.sizeOfRecordTargetArray() == 1) {
+            POCDMT000002UK01PatientRole patientRole = clinicalDocument1.getRecordTargetArray(0).getPatientRole();
+            Patient patient = patientMapper.mapPatient(patientRole);
+            encounter.setSubject(new Reference(patient));
+            encounter.setSubjectTarget(patient);
+        } else if (clinicalDocument1.sizeOfRecordTargetArray() > 1){
+            Group group = groupMapper.mapGroup(clinicalDocument1.getRecordTargetArray());
+            Group.GroupMemberComponent groupMemberComponent = new Group.GroupMemberComponent();
+            groupMemberComponent.setEntity(groupMemberComponent.getEntity());
```

Comment:
groupMemberComponent.setEntity(groupMemberComponent.getEntity()); -> are you sure this should be like that?You take entity from groupMemberComponent to set it again.... Please correct it

Reply:
I think this like could be simply removed

Reply:
Removed parsing of members (GroupMemberComponent).